src/proxy/processors/push-action/checkSensitiveData.js

```diff
@@ -0,0 +1,136 @@
+        let sensitiveDataFound = false;
+        try {
+            const workbook = XLSX.readFile(filePath);
+            const sheetName = workbook.SheetNames[0];
+            const sheet = workbook.Sheets[sheetName];
+            const jsonData = XLSX.utils.sheet_to_json(sheet);
+            jsonData.forEach((row) => {
+                for (const [key, value] of Object.entries(row)) {
+                    if (checkForSensitiveData(value)) {
+                        console.log(`\x1b[33mSensitive data found in XLSX: ${key}: ${value}\x1b[0m`); // Log in yellow
+                        sensitiveDataFound = true;
+                    }
+                }
+            });
+            if (!sensitiveDataFound) {
+                console.log('No sensitive data found in XLSX.');
+            }
+            resolve(sensitiveDataFound); // Resolve with the flag indicating if sensitive data was found
+        } catch (error) {
+            console.error(`Error reading XLSX file: ${error.message}`);
+            reject(error); // Reject the promise on error
+        }
+    });
+};
+// Function to check for sensitive data in .log and .json files
+const checkLogJsonFiles = async (filePath) => {
+    return new Promise((resolve, reject) => {
+        let sensitiveDataFound = false;
+        fs.readFile(filePath, 'utf8', (err, data) => {
+            if (err) {
+                console.error(`Error reading file ${filePath}: ${err.message}`);
+                return reject(err);
+            }
+            if (checkForSensitiveData(data)) {
+                console.log(`\x1b[33mSensitive data found in ${filePath}\x1b[0m`);
+                sensitiveDataFound = true;
+            }
+            resolve(sensitiveDataFound);
+        });
+    });
+};
+// Function to parse the file based on its extension
+const parseFile = async (filePath) => {
+    const ext = path.extname(filePath).toLowerCase();
+    
+    switch (ext) {
+        case '.csv':
+            return await processCSV(filePath);
+        case '.xlsx':
+            return await processXLSX(filePath);
+        case '.log':
+            return await checkLogJsonFiles(filePath);
+        case '.json':
+            return await checkLogJsonFiles(filePath);
+        default:
+            // Skip unsupported file types without logging
+            return false; // Indicate that no sensitive data was found for unsupported types
+    }
+};
+// Async exec function to handle actions
+const exec = async (req, action) => {
+    // getDiffExec(req, action); // Call to getDiffExec if necessary
+    const diffStep = action.steps.find((s) => s.stepName === 'diff');
+    if (diffStep && diffStep.content) {
+        console.log('Diff content:', diffStep.content);
+        const filePaths = diffStep.content.filePaths || [];
+        if (filePaths.length > 0) {
+            // Check for sensitive data in all files
+            const sensitiveDataFound = await Promise.all(filePaths.map(parseFile));
```

Comment:
shouldn't we prioritize concurrent execution over resource allocation as if the no of files increases it would slow down the speed if we go with for .. of with await approach although I agree in case large datasets it would consume a lot of memory.